src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -298,6 +298,24 @@ protected final void verifyWithInlineConfigParser(String filePath1,
             filePath2, expectedFromFile2));
     }
 
+    /**
+     * Performs verification of the file with the given file path using specified configuration
+     * and the array of expected messages. Unlike verifyWithInlineConfigParser, this method
+     * does not perform an additional verification step, thereby executing the Check only once.
+     *
+     * @param filePath file path to verify.
+     * @param expected an array of expected messages.
+     * @throws Exception if exception occurs during verification process.
+     */
+    protected final void verifyOnceWithInlineConfigParser(String filePath, String... expected)
```

Comment:
As it is performance issue.
We don't need new method, we just need to fix existing method and fix it for all executions, all tests.
I hope in final result you archive significant performance improvement of ` mvn test` so reduce significantly whole CI execution, and execution on local

Reply:
Do we need to add an overloaded method or do we need to modify the content of existing method only. Also, modifying it for one case might not suit the other case imo. We can try if that's the requirement.

Reply:
Let's try migrate all at one PR, they should work the same way, you can reuse existing method and simply optimize it, no overloaded methods are required.